### PR TITLE
Task 1: Cargo [operator==]

### LINKED ISSUE #1: Task 1: Cargo  [operator==]
W klasie Cargo napisz operator porównania (operator==), który będzie sprawdzał, czy towary są takie same.

### PR DESCRIPTION
**Pull request by lksobczyk**

### CODE REVIEW

File: shm/Cargo.cpp [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+    _amount -= amount;
     return *this;
 }
+
+bool Cargo::operator==(const Cargo& src) const {
+     return (_name == src._name) && (_amount == src._amount) && (_basePrice == src._basePrice);
+}
+
+bool Cargo::operator!=(const Cargo& src) const {
+    return !operator==(src);
```
Comment (kszytko) [inline shm/Cargo.cpp]:
```suggestion
    return !(this == src);
```
In my opinion this is better solution. What do you think?

File: shm/Cargo.hpp [hunk 2]
```diff
@@ -1,12 +1,17 @@
...
 
+#include <string>
+
 class Cargo {
 public:
     Cargo& operator+=(const size_t);
     Cargo& operator-=(const size_t);
 
+    bool operator==(const Cargo&) const;
+    bool operator!=(const Cargo&) const;
```
Comment (kszytko) [inline shm/Cargo.hpp]: We should add the names of function argumets. While reading header you just now what exacly this function does.
```c++
    Cargo& operator+=(const size_t amount);
    Cargo& operator-=(const size_t amount);
```

File: shm/Makefile [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	$(CC) $(OBJECTS) -o $(PROGRAM)
+
+run:  $(PROGRAM)
+	@echo Running program ...
+	./$(PROGRAM)
+
+clean:
+	rm -f $(OBJECTS) $(PROGRAM)
+
+.PHONY: all clean
```
Comment (kszytko) [inline shm/Makefile]: **AWESOME**

File: shm/Ship.hpp [hunk 4]
```diff
@@ -1,28 +1,30 @@
...
+    size_t getCapacity() const { return _capacity; }
+    size_t getCrew() const { return _crew; }
 
-    void setName(const std::string& name) { m_name = name; }
+    void setName(const std::string& name) { _name = name; }
 
-    Ship& operator+=(const int);
-    Ship& operator-=(const int);
+    Ship& operator+=(const size_t);
+    Ship& operator-=(const size_t);
```
Comment (kszytko) [inline shm/Ship.hpp]: Add arguments here as well please.

**Review by kszytko — CHANGES_REQUESTED** [review]
Some small changes to make.

**Review by madebykj — CHANGES_REQUESTED** [review]
kszytko comments.

File: shm/Cargo.cpp (hunk 1 above)
Comment (madebykj) [inline shm/Cargo.cpp]: Do we need both operators (== and !=)?
I think that operator == is enough.

Comment (lksobczyk) [reply shm/Cargo.cpp]: I know that only one operator should be added in the task. In my opinion these operators are symmetrical, so I added both. Similarly to the rule of zero/three/five. Some programmers also add operators of relation "<" and ">".

Comment (madebykj) [reply shm/Cargo.cpp]: OK, I don't know those rules yet.

**Review by madebykj — APPROVED** [review]

**Review by kszytko — APPROVED** [review]

**Pull request merged**